### PR TITLE
AAE-11760 git-latest-tag-sha action

### PR DESCRIPTION
**Pull request by ozeta**
this action finds the latest tag for a repository, and returns its long sha

### CODE REVIEW

File: .github/actions/git-latest-tag-sha/action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      run: |
+        if [ -n "$REPO_DIR" ]
+        then
+          cd $REPO_DIR
+        fi
+        git fetch --tags
+        last_tag=$(git tag --sort=-creatordate | head -n 1)
+        tag_long_sha=$(git rev-list -n 1 $last_tag)
+        echo "Long SHA for tag $last_tag is $tag_long_sha"
+        echo "tag_long_sha=$tag_long_sha" >> $GITHUB_OUTPUT
```
Comment (gionn) [inline .github/actions/git-latest-tag-sha/action.yml:24]: no need to duplicate the existing git-latest-tag, just extend that by adding a new output

**Review by gionn — CHANGES_REQUESTED** [review]

File: .github/actions/git-latest-tag/tests/git-latest-tag.bats [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    export GITHUB_HEAD_REF="AAE-11760-get-latest-tag-sha"
+
+    # Mock git-latest-tag defaults
+    export PATTERN="*"
+    export REPO_DIR="$PWD"
+}
+
+@test "latest_tag" {
+    latest_tag=$(git tag --sort=-creatordate | head -n 1)
+    tag_sha=$(git rev-list -n 1 $latest_tag)
```
Comment (gionn) [inline .github/actions/git-latest-tag/tests/git-latest-tag.bats:19]: it doesn't make sense to use in the test the same code used in the code under test to check the behavior.

you could verify that the message is structured correctly and do not check any specific tag/sha (that will change every time a new release is made)

**Review by gionn — CHANGES_REQUESTED** [review]

File: .github/actions/git-latest-tag/tests/git-latest-tag.bats (hunk 2 above)
Comment (ozeta) [reply .github/actions/git-latest-tag/tests/git-latest-tag.bats:19]: yeah i tought that was an unuseful test and actually I was wait for a suggestion 😂

File: .github/actions/git-latest-tag/tests/git-latest-tag.bats [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    # Mock git-latest-tag defaults
+    export PATTERN="*"
+    export REPO_DIR="$PWD"
+}
+
+@test "latest_tag" {
+    regex="v[0-9]+\.[0-9]+\.[0-9]+"
+    run git-latest-tag.sh
+    [ "$status" -eq 0 ]
+    if [[ ! $output =~ $regex ]]; then
```
Comment (gionn) [inline .github/actions/git-latest-tag/tests/git-latest-tag.bats:21]: I would have checked the entire string via regex but there is also the other test so in the end is fine for me

**Review by gionn — APPROVED** [review]

**Pull request merged**